backend/app/request-validation.js

```diff
@@ -0,0 +1,76 @@
+function stringValidationFn (value) {
+  return typeof(value) === "string"
+}
+
+function intValidationFn (value) {
+  return Number.isInteger(value)
+}
+
+function completionFrequencyValidationFn (value) {
+  return (typeof(value) === "string" && ["E1D", "E2D", "E7D"].includes(value))
+}
+
+/*
+ * The TypeValidator data structure maps an identifying data type to be 
+ * included in a request body to the function that validates it. The
+ * validation function must take in a single argument (the value it's
+ * validating) and return a boolean indicating the value's validity.
+ */
+const TypeValidator = {
+  STRING: stringValidationFn,
+  INT: intValidationFn,
+  COMPLETION_FREQUENCY: completionFrequencyValidationFn,
+}
+
+/*
+ * The RequestStructure class is a collection of RequestParameter's that
+ * encapsulates the expected structure of data included in the request.
+ */
+class RequestStructure {
+  constructor(...requestParameters) {
+    this.params = requestParameters
+  }
+
+  validate(requestBody) {
+    let errorObj = {}
+    for (const param of this.params) {
+      const errorMessage = param.validate(requestBody)
+      if (errorMessage) {
+        errorObj[param.name] = errorMessage
+      }
+    }
+    return errorObj
```

Comment:
I think it might make sense to keep a flag that checks if there were any errors and return it as an array like: `return [valid, errorObj]` where `valid` is the boolean flag. This way we only have to process the errorObj if there were any errors and it is a bit more explicit about validity